StRoot/StFwdTrackMaker/include/Tracker/TrackFitter.h

```diff
@@ -0,0 +1,931 @@
+#ifndef TRACK_FITTER_H
+#define TRACK_FITTER_H
+
+#include "GenFit/ConstField.h"
+#include "GenFit/EventDisplay.h"
+#include "GenFit/Exception.h"
+#include "GenFit/FieldManager.h"
+#include "GenFit/KalmanFitStatus.h"
+#include "GenFit/KalmanFitter.h"
+#include "GenFit/KalmanFitterInfo.h"
+#include "GenFit/KalmanFitterRefTrack.h"
+#include "GenFit/MaterialEffects.h"
+#include "GenFit/PlanarMeasurement.h"
+#include "GenFit/RKTrackRep.h"
+#include "GenFit/SpacepointMeasurement.h"
+#include "GenFit/StateOnPlane.h"
+#include "GenFit/TGeoMaterialInterface.h"
+#include "GenFit/Track.h"
+#include "GenFit/TrackPoint.h"
+
+#include "Criteria/SimpleCircle.h"
+
+#include "TDatabasePDG.h"
+#include "TGeoManager.h"
+#include "TMath.h"
+#include "TRandom.h"
+#include "TRandom3.h"
+#include "TVector3.h"
+
+#include <vector>
+
+#include "StFwdTrackMaker/XmlConfig/HistoBins.h"
+#include "StFwdTrackMaker/XmlConfig/XmlConfig.h"
+#include "StFwdTrackMaker/include/Tracker/FwdHit.h"
+#include "StFwdTrackMaker/include/Tracker/STARField.h"
+#include "StFwdTrackMaker/include/Tracker/loguru.h"
+#include "StFwdTrackMaker/include/Tracker/FwdGeomUtils.h"
+
+
+class TrackFitter {
+
+  public:
+    TrackFitter(jdb::XmlConfig &_cfg) : cfg(_cfg) {
+        fTrackRep = 0;
+        fTrack = 0;
+    }
+
+    void setup(bool make_display = false) {
+        LOG_SCOPE_FUNCTION(INFO);
+
+        TGeoManager * gMan = nullptr;
+
+        {
+            LOG_SCOPE_F( INFO, "Setup Geometry in GENFIT" );
+            // gMan = new TGeoManager("Geometry", "Geane geometry");
+            TGeoManager::Import(cfg.get<string>("Geometry", "fGeom.root").c_str());
+            gMan = gGeoManager;
+            genfit::MaterialEffects::getInstance()->init(new genfit::TGeoMaterialInterface());
+            genfit::MaterialEffects::getInstance()->setNoEffects(cfg.get<bool>("TrackFitter::noMaterialEffects", false));
+            if ( cfg.get<bool>("TrackFitter::noMaterialEffects", false) ){
+                LOG_F( WARNING, "MaterialEffects are turned OFF" );
+            }
+        }
+
+        // TODO : Load the STAR MagField
+        genfit::AbsBField *bField = nullptr;
+
+        if (0 == _gField) {
+            if (cfg.get<bool>("TrackFitter:constB", false)) {
+                bField = new genfit::ConstField(0., 0., 5.);
+                LOG_F(INFO, "Using a CONST B FIELD");
+            } else {
+                bField = new genfit::STARFieldXYZ();
+                LOG_F(INFO, "Using STAR B FIELD");
+            }
+        } else {
+            LOG_F(INFO, "Using StarMagField interface");
+            bField = _gField;
```

Comment:
Why do you need to use a global pointer to setup the local variable bField? Can't you just use the StarFieldAdaptor directly?